frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -74,13 +76,34 @@ const CloudShellExec: React.FC<CloudShellExecProps> = ({
 
   const tick = useActivityTick(workspaceName, namespace);
 
-  const onData = React.useCallback(
-    (data: string): void => {
-      tick();
-      ws.current?.send(`0${Base64.encode(data)}`);
-    },
-    [tick],
-  );
+  React.useEffect(() => {
+    let tickOnInterval = setInterval(tick, 30000);
+
+    // Function to handle ticking
+    const handleTickOnVisible = () => {
+      if (document.visibilityState === 'hidden' || !isActiveTab) {
+        clearInterval(tickOnInterval);
+        tickOnInterval = null;
+      }
+      if (document.visibilityState === 'visible' && isActiveTab) {
+        tickOnInterval = tickOnInterval || setInterval(tick, 60000);
+        tick(); // Start with  a sync tick
```

Comment:
Why do you need to call tick() seperately

Reply:
updated